server/web/context/context.go

```diff
@@ -77,6 +78,91 @@ type Context struct {
 	_xsrfToken     string
 }
 
+type StreamEvent struct {
+	Content          string
+	ReasoningContent string
+	Error            error
+	Done             bool
+}
+
+func (ctx *Context) EventStreamResp() chan<- StreamEvent {
+	eventCh := make(chan StreamEvent)
+	go func() {
+		for {
+			select {
+			case event, ok := <-eventCh:
+				if !ok || event.Done {
+					endEvent(ctx)
+					return
+				}
+
+				if event.Error != nil {
+					errorEvent(ctx, event.Error)
+					return
+				}
+				msgEvent(ctx, event)
+			case <-ctx.Request.Context().Done():
```

Comment:
Please confirm whether this code path will be reached if the caller forgot to close the channel. if so, I think we don't need to require the caller must close the channel.